lnrpc/routerrpc/router_backend.go

```diff
@@ -1158,7 +1158,21 @@ func (r *RouterBackend) MarshallPayment(payment *channeldb.MPPayment) (
 		PaymentRequest:  string(payment.Info.PaymentRequest),
 		Status:          status,
 		Htlcs:           htlcs,
-	}, nil
+		FailureReason:   lnrpc.Payment_NONE,
+	}
+
+	// Marshall failure reason if any.
+	if payment.FailureReason != nil {
```

Comment:
style nit: do this above, before creating the final `lnrpc.Payment`

Reply:
fixed